### PR TITLE
Should not display FSType in describe pv for AWS EFS driver

### PR DESCRIPTION
**Pull request by RomanBednar**
**Is this a bug fix or adding new feature?**
This is a bug fix but it changes how StorageClass is checked. StorageClass objects with invalid fstype capability will no longer pass the check and volume creation will be prevented. This applies to **all new PVs created** - similar to other checks that were added in the past (e.g. [access mode](https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/fc4d43d0f6ac3ff6619672c9862ef88b31ab3506/pkg/driver/node.go#L318)).

**What is this PR about? / Why do we need it?**
This change is needed to prevent PVs being created with incorrect values, currently the driver let's users create PVs with any value which is not correct. We should check if supported fstype was used.

**What testing is done?** 
Adding unit tests to cover the new check, both negative and positive flow. Also added a test for multiple messages which can happen when there are multiple unsupported capabilities found.

### CODE REVIEW

**Comment by RomanBednar** [comment]
The driver should now accept only "efs", "nfs, "" values for fstype in storage class. If incorrect value is used provisioning will fail.

Here is what I used for testing the patch manually:


1. StorageClass with incorrect "fstype: abc"

$ oc get sc/efs-sc-fstype -o yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: "2022-01-12T14:02:15Z"
  name: efs-sc-fstype
  resourceVersion: "53362"
  uid: 938c14c9-2552-461d-8497-eea4fe5f7af1
mountOptions:
- tls
parameters:
  FStype: abc
  basePath: /dynamic_provisioning
  directoryPerms: "700"
  fileSystemId: fs-0f3a7db6f271f92e6
  provisioningMode: efs-ap
provisioner: efs.csi.aws.com
reclaimPolicy: Delete
volumeBindingMode: Immediate


2. Claim that references the incorrect StorageClass

$ oc get pvc/efs-claim-fail -o yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    volume.beta.kubernetes.io/storage-provisioner: efs.csi.aws.com
  creationTimestamp: "2022-01-12T14:17:58Z"
  finalizers:
  - kubernetes.io/pvc-protection
  name: efs-claim-fail
  namespace: default
  resourceVersion: "58155"
  uid: 22828df9-c5e2-4907-9c45-8cf4a170529d
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: 5Gi
  storageClassName: efs-sc-fstype
  volumeMode: Filesystem
status:
  phase: Pending


3. Verify new error message about incorrect fstype

$ oc describe pvc/efs-claim-fail
Name:          efs-claim-fail
Namespace:     default
StorageClass:  efs-sc-fstype
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-provisioner: efs.csi.aws.com
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Used By:       efs-app-fail
Events:
  Type     Reason                Age                  From                                                                  Message
  ----     ------                ----                 ----                                                                  -------
  Normal   Provisioning          20s (x8 over 2m27s)  efs.csi.aws.com_ip-10-0-149-254_b82e2156-1754-432b-92d1-739c1ddb4480  External provisioner is provisioning volume for claim "default/efs-claim-fail"
  Warning  ProvisioningFailed    20s (x8 over 2m27s)  efs.csi.aws.com_ip-10-0-149-254_b82e2156-1754-432b-92d1-739c1ddb4480  failed to provision volume with StorageClass "efs-sc-fstype": rpc error: code = InvalidArgument desc = Volume capabilities not supported: invalid fstype: abc
  Normal   ExternalProvisioning  2s (x11 over 2m27s)  persistentvolume-controller                                           waiting for a volume to be created, either by external provisioner "efs.csi.aws.com" or manually created by system administrator

**Comment by jsafrane** [comment]
> The driver should now accept only "efs", "nfs", "" values

@wongma7 @nckturner what do you think? Do you consider "efs" as the only `fsType` or  is "nfs" valid too?
Note that this is only about validation, choosing `nfs` won't disable encryption.

The code LGTM.

**Comment by wongma7** [comment]
I think only efs should be accepted. Otherwise lgtm.

i dont' want to give impression that there is a difference between fstype nfs and fstype efs. sure, under the hood, mount.efs is just calling mount.nfs so maybe they should be treated the same..but i think it's simpler to just accept efs nd nothing else. Can loosen this restriction later if somebody complains!

**Comment by RomanBednar** [comment]
Sounds good, removing nfs as suggested. Thank you for your feedback Matthew.

File: pkg/driver/node.go [hunk 1]
```diff
@@ -36,7 +36,8 @@ var (
 		csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
 		csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER,
 	}
-	volumeIdCounter = make(map[string]int)
+	volumeIdCounter  = make(map[string]int)
+	supportedFsTypes = []string{"efs", ""}
```
Comment (jsafrane) [inline pkg/driver/node.go:40]: nit: `supportedFSTypes`. Abbreviations have all characters in uppercase.
